frontend/src/logic/offerlogic.ts

```diff
@@ -61,41 +61,48 @@ const offerConverter = Object.freeze({
     description: description,
     shopLocation: shopLocation,
     expectedDeliveryTime: expectedDeliveryTime,
-    status: Status[status],
+    status: Status[status]
   }),
   fromFirestore: (offerSnapshot: firebase.firestore.DocumentSnapshot) => {
     const data = offerSnapshot.data();
     if (data === undefined) {
       throw new Error("Unable to find snapshot for offer.");
     }
+
+    const expectedDeliveryTime: number = data.expectedDeliveryTime;
+    const status: Status = Status[data.status];
+    const newStatus = shouldShowExpired(expectedDeliveryTime, status) ? Status.EXPIRED : status;
+    
     return new Offer(
       data.uid,
       offerSnapshot.id,
       data.title,
       data.description,
       data.shopLocation,
-      data.expectedDeliveryTime,
-      Status[data.status],
+      expectedDeliveryTime,
+      newStatus
     );
   },
 });
 
 /**
- * Gets open offer
+ * Gets open offers.
  */
-export const getOpenOffer: () => Promise<Offer[]> = async () => {
+export const getOpenOffers: () => Promise<Offer[]> = async () => {
   const offers: Offer[] = [];
-  const requestRef = await db.collection(COLLECTION_OFFERS).get();
+  const offersRef = await db.collection(COLLECTION_OFFERS)
+    .where("status", "==", Status.OPEN)
+    .get();
   const temp: firebase.firestore.QueryDocumentSnapshot<
     firebase.firestore.DocumentData
   >[] = [];
-  requestRef.forEach((offerSnapshot) => {
+  offersRef.forEach((offerSnapshot) => {
     temp.push(offerSnapshot);
   });
   await Promise.all(
     temp.map(async (offerSnapshot) => {
       try {
-        const offer = await offerConverter.fromFirestore(offerSnapshot);
+        const offer = offerConverter.fromFirestore(offerSnapshot);
```

Comment:
this is not an async method for offer is it? Cause i recall for request we still need to make a call to get the task

Reply:
Nope. It isn't. 😅